packages/adblocker/src/engine/engine.ts

```diff
@@ -229,6 +229,85 @@ export default class FilterEngine extends EventEmitter<EngineEventHandlers> {
     return engine as InstanceType<T>;
   }
 
+  public static fromEngines<T extends FilterEngine>(
+    this: new (...args: any[]) => T,
+    ...engines: T[]
+  ): T {
+    if (engines.length === 0) {
+      throw new Error(`merging engines requires at least two engines, use ".empty()" instead`);
+    }
+
+    const config = engines[0].config;
+    const lists = engines[0].lists;
+    const version = engines[0].version;
+
+    const networkFilters: Map<number, NetworkFilter> = new Map();
+    const cosmeticFilters: Map<number, CosmeticFilter> = new Map();
+    const preprocessors: Preprocessor[] = [];
+
+    const configKeysMustMatch: Exclude<keyof Config, 'serialize' | 'getSerializedSize'>[] = [
+      'debug',
+      'enableCompression',
+      'enableOptimizations',
+    ];
+
+    for (const engine of engines) {
+      // Validate the config
+      for (const configKey of configKeysMustMatch) {
+        if (config[configKey] !== engine.config[configKey]) {
+          throw new Error(
+            `config "${configKey}" is not compatible with base engine and cannot be converted`,
```

Comment:
it is not clear what a base engine is. we can say something like `configs of all merged engines must be the same`